examples/index.tsx

```diff
@@ -44,4 +45,5 @@ export default class App extends React.Component {
   }
 }
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const root = ReactDOM.createRoot(document.getElementById('root'));
```

Comment:
This will fail, since our `devDependencies` specifies `^16.8.0` for react, right?

Reply:
That's true. Should I revert the change for compatability with 16 or upgrade the dev dependency to 18? I'm leaning towards the latter.

I can upgrade the #14 CR to run a matrix of React versions during CI to make sure tests pass with every (major) React version.

Reply:
I like that idea.  Let's update the devDep, and at least run two react versions in CI, the oldest we support, and the latest 18.x.